sdk/java/pulumi/src/main/java/com/pulumi/resources/StackReferenceArgs.java

```diff
@@ -7,26 +7,79 @@
 import java.util.Optional;
 
 /**
- * The set of arguments for constructing a StackReference resource.
+ * The set of arguments for constructing a {@link StackReference} resource.
  */
-public class StackReferenceArgs extends ResourceArgs {
-
-    public static final StackReferenceArgs Empty = new StackReferenceArgs(null) {
-        // Empty
-    };
+public final class StackReferenceArgs extends ResourceArgs {
 
     /**
-     * The name of the stack to reference.
+     * An empty {@link StackReferenceArgs} instance.
      */
+    public static final StackReferenceArgs Empty = StackReferenceArgs.builder().build();
+
     @Import(name = "name", required = true)
     @Nullable
-    public final Output<String> name;
+    private final Output<String> name;
 
     public StackReferenceArgs(@Nullable Output<String> name) {
         this.name = name;
     }
 
+    /**
+     * The name of the stack to reference.
+     *
+     * @deprecated use {@link #name()}
+     */
+    @Deprecated
     public Optional<Output<String>> getName() {
+        return name();
+    }
+
+    /**
+     * The name of the stack to reference.
+     */
+    public Optional<Output<String>> name() {
         return Optional.ofNullable(name);
     }
+
+    /**
+     * @return a {@link Builder} instance
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * The builder for {@link StackReferenceArgs}.
+     */
+    public static final class Builder {
+        @Nullable
+        private Output<String> name;
+
+        /**
+         * @param name the name of the stack to reference.
+         * @return the {@link Builder}
+         * @see StackReferenceArgs#name()
+         */
```

Comment:
Nit: would like something describing the structure of the string `<organization>/<project>/<stack>`

Reply:
It's actually <stack> or <org>/<stack> or <org>/<proj>/<stack> just to complicate things :)